docs/source/deployment/lambda.mdx

```diff
@@ -0,0 +1,387 @@
+import { ApolloServer } from '@apollo/server';
+import { startServerAndCreateLambdaHandler } from '@as-integrations/aws-lambda'; //highlight-line
+
+const typeDefs = `#graphql
+  type Query {
+    hello: String
+  }
+`;
+
+const resolvers = {
+  Query: {
+    hello: () => 'world',
+  },
+};
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+});
+
+// This final export is important!
+export const graphqlHandler = startServerAndCreateLambdaHandler(server); //highlight-line
+```
+
+```js title="server.js"
+import { ApolloServer } from '@apollo/server';
+import { startServerAndCreateLambdaHandler } from '@as-integrations/aws-lambda'; //highlight-line
+
+const typeDefs = `#graphql
+  type Query {
+    hello: String
+  }
+`;
+
+const resolvers = {
+  Query: {
+    hello: () => 'world',
+  },
+};
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+});
+
+// This final export is important!
+export const graphqlHandler = startServerAndCreateLambdaHandler(server); //highlight-line
+```
+
+</MultiCodeBlock>
+
+The final line in the code snippet above creates an export named `graphqlHandler` with a Lambda function handler. We'll get back to this function in a moment!
+
+## Deploying using the Serverless framework
+
+[Serverless](https://serverless.com) is a framework that helps make deploying serverless applications to platforms like AWS Lambda easier.
+
+### Installing the CLI
+
+We'll use the [Serverless CLI](https://www.serverless.com/framework/docs/getting-started/) to deploy our application. You can either install the Serverless package into your project directly or install the Serverless CLI globally:
+
+```bash
+npm install -g serverless
+```
+
+The Serverless CLI can access the credentials of the AWS CLI, which [you configured earlier](#prerequisites). So now we just need to tell Serverless which service we want to deploy.
+
+> AWS best practices recommend [rotating your access keys](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#rotate-credentials) for use cases that require long-term credentials (e.g., hosting an application).
+
+### Configuring services
+
+You can configure Serverless using a `serverless.yml` file, letting it know which services to deploy and where the handlers are.
+
+If you are using TypeScript, download the `serverless-plugin-typescript` package to enable Serverless to use your TS file:
+
+```bash
+npm install -D serverless-plugin-typescript
+```
+
+You use the example `serverless.yml` configuration below; take care to ensure the file path you use is pointing to the file where you export your handler:
+
+```yaml title="serverless.yml"
+service: apollo-lambda
+provider:
+  name: aws
+  runtime: nodejs16.x
+functions:
+  graphql:
+    # Make sure your file path is correct!
+    # (e.g., if your file is in the root folder use server.graphqlHandler )
+    # The format is: <FILENAME>.<HANDLER>
+    handler: src/server.graphqlHandler # highlight-line
+    events:
+      - http:
```

Comment:
If I understand [the docs](https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml#api-gateway-v2-http-api-1) correctly, this is enabling API Gateway v1. Maybe makes sense to do v2 instead? Not an expert on the difference, just assume the newer one is better.

Reply:
Great catch! Updated to using v2:

```yaml title="serverless.yml"
service: apollo-lambda
provider:
  name: aws
  runtime: nodejs16.x
  httpApi:
    cors: true
functions:
  graphql:
    # Make sure your file path is correct!
    # (e.g., if your file is in the root folder use server.graphqlHandler )
    # The format is: <FILENAME>.<HANDLER>
    handler: src/server.graphqlHandler # highlight-line
    events:
      - httpApi:
          path: /
          method: POST
      - httpApi:
          path: /
          method: GET
# Omit the following lines if you aren't using TS!
plugins:
  - serverless-plugin-typescript
```